toolchest_client/api/urls.py

```diff
@@ -8,12 +8,41 @@
 
 import os
 
-# Base URLs used by the server API.
-BASE_URL = os.environ.get("BASE_URL", "https://api.toolche.st")
 
-PIPELINE_SEGMENT_INSTANCES_ROUTE = "/pipeline-segment-instances"
-PIPELINE_SEGMENT_INSTANCES_URL = BASE_URL + PIPELINE_SEGMENT_INSTANCES_ROUTE
+def get_api_url():
+    """Retrieves the base URL for the Toolchest server API. Defaults to "https://api.toolche.st"
+    if a custom API URL is not set.
+    """
+    # Note: BASE_URL is checked for backwards compatibility.
+    return os.environ.get("TOOLCHEST_API_URL", os.environ.get("BASE_URL", "https://api.toolche.st"))
 
-S3_ROUTE = "/s3"
-S3_URL = BASE_URL + S3_ROUTE
-S3_METADATA_URL = S3_URL + "/metadata"
+
+def get_pipeline_segment_instances_url():
+    """Retrieves the Toolchest API Route for pipeline segment instances. Used internally."""
+    PIPELINE_SEGMENT_INSTANCES_ROUTE = "/pipeline-segment-instances"
+    return get_api_url() + PIPELINE_SEGMENT_INSTANCES_ROUTE
+
+
+def get_s3_metadata_url():
+    """Retrieves the Toolchest API Route for S3 metadata. Used internally."""
+    S3_ROUTE = "/s3"
+    S3_URL = get_api_url() + S3_ROUTE
+    return S3_URL + "/metadata"
+
+
+def set_api_url(custom_api_url=None):
+    """Sets the Toolchest API URL (env var TOOLCHEST_API_URL) to the given value.
+    If a URL is not provided, resets to the default Toolchest API URL.
+
+    :param custom_api_url: Custom API URL. Any trailing slashes should be removed.
+
+    Usage::
+
+        >>> import toolchest_client as toolchest
+        >>> toolchest.set_api_url("http://your.custom.api.url.here")
+
+    """
+    if custom_api_url:
+        os.environ["TOOLCHEST_API_URL"] = custom_api_url
+    else:
+        os.environ.pop("TOOLCHEST_API_URL", None)
```

Comment:
What's the intent here?

Reply:
Wouldn't the same be accomplished by the other block of the if/else?

Reply:
Not exactly, since the key `TOOLCHEST_API_URL` would still exist in `os.environ` if something like `pop` wasn't called. (In fact, for any `os.environ["variable_name"] = x` call, `x` has to be a string, or Python will error.)

The `pop` call clears the env var key to avoid all future calls of `get_api_url` returning something like `None`. (It does feel inelegant, but there aren't any better options, according to StackOverflow.)

Reply:
Gotcha, makes sense!